Remove a lot of mentions of JavaScript versions (#19448)
mdn/content: files/en-us/web/javascript/reference/iteration_protocols/index.md

---
{{jsSidebar("More")}}

> **Iteration protocols** aren't new built-ins or syntax, but _protocols_. These protocols can be implemented by any object by following some conventions.

Reviewer: This intro is really strange:
- It starts by explaining what it isn't.
- The use of _new_. It isn't new anymore.

What about something like.
```suggestion
**Iteration protocols** are _algorithms_ that can be implemented by any object following some conventions.
```

Author: Any ES feature can be categorized into "syntax", "API", or "protocol". Well-known symbols are protocols, so are thenables and iteration protocols. I find the distinguishing here understandable (though could be a bit weird especially with "new")
